### PR TITLE
Fixed wrong printing of pow when base is Mul

### PR DESCRIPTION
**Pull request by shivamvats**

### CODE REVIEW

File: src/pow.cpp [hunk 1]
```diff
@@ -92,6 +92,8 @@ std::string Pow::__str__() const
     std::ostringstream o;
     if (is_a<Add>(*base_)) {
         o << "(" << *base_ << ")";
+    } else if (is_a<Mul>(*base_)) {
+        o << "(" << *base_ << ")";
```
Comment (thilinarmtb) [inline src/pow.cpp]: We can do the following, right?
```
if (is_a<Add>(*base_) || is_a<Mul>(*base_)) {
         o << "(" << *base_ << ")";
}
```

Comment (shivamvats) [inline src/pow.cpp]: We definitely can. But I felt having separate conditions for different classes will make it easier if we need to add subconditions to any of the classes, in future.
On second thought, however, we can make the changes when the need arises.

Comment (isuruf) [inline src/pow.cpp]: This may be unrelated to the PR, but we should also check for `Pow`, since `(x**y)**z` prints as `x^y^z`

Comment (thilinarmtb) [inline src/pow.cpp]: What subconditions are you thinking about?

Comment (certik) [inline src/pow.cpp]: Indeed, why not to write this as:
```c++
if (is_a<Add>(*base_) || is_a<Mul>(*base_) || is_a<Pow>(*base_)) {
         o << "(" << *base_ << ")";
}
```
And add tests for `(x**y)**z` as well.

Comment (shivamvats) [inline src/pow.cpp]: @thilinarmtb I can't think of any sub-condition now. I assumed that we might have them in future. 
I'm making appropriate changes and adding the case for pow.

**Comment by certik** [comment]
+1 to merge if tests pass.

**Comment by isuruf** [comment]
@shivamvats thanks for the PR

**Comment by thilinarmtb** [comment]
@shivamvats:  Thanks for the fix !!!

**Pull request merged**